modules/javafx.graphics/src/main/native-glass/mac/GlassMenu.m

```diff
@@ -262,6 +262,22 @@ - (void)_setPixels:(jobject)pixels
         (*env)->CallVoidMethod(env, pixels, jPixelsAttachData, ptr_to_jlong(&image));
         if (image != NULL)
         {
+            jclass pixelsClass = (*env)->FindClass(env, "com/sun/glass/ui/mac/MacPixels");
+
+            jfieldID jPixelsWidthField = (*env)->GetFieldID(env, pixelsClass, "width", "I");
```

Comment:
After each JNI call you need to check for exceptions like:
```
jfieldID jPixelsWidthField = (*env)->GetFieldID...
if ((*env)->ExceptionCheck(env)) return;
```

Reply:
Thanks @jperedadnr.
The last line inside that if block is `[image release]`. I am assuming that this is a memory cleanup from `(*env)->CallVoidMethod(env, pixels, jPixelsAttachData, ptr_to_jlong(&image));`. 

If I am adding the `ExceptionCheck` after each JNI call, should I make my block:-
```
jfloat scaley = (*env)->GetFloatField(env, pixels, jPixelsScaleYField);
if ((*env)->ExceptionCheck(env)) {
    [image release];
    return;
}
```

Reply:
You can apply the checks only when you get the jclass and jfields in `initIDs`. 
Then you can simply add an if that verifies that jPixelsXXXField > 0 before getting the jint/jfloat values.